api-mock/utils.ts

```diff
@@ -0,0 +1,68 @@
+import { minify_sync } from "terser";
+import fs = require("fs");
+import SuperTokens from "supertokens-node";
+import { MockedVars } from "./api-mock-server";
+import { User as UserClass } from "supertokens-node/lib/build/user";
+
+const uniqueFn = new Set<string>();
+
+export function minify(code: string) {
+    const result = minify_sync(code, {
+        compress: {
+            drop_console: true,
+            expression: true,
+            keep_fnames: true,
+        },
+    });
+
+    // TODO: remove this once we have all functions that we need to handle
+    if (result.code) {
+        uniqueFn.add(result.code);
+        fs.writeFileSync("overridesFn.json", JSON.stringify(Array.from(uniqueFn).sort(), null, 2));
+    }
+    return result.code;
+}
+
+export function randomString(length = 30) {
+    let result = "";
+    let characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    let charactersLength = characters.length;
+    for (let i = 0; i < length; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
+    return result;
+}
+
+export function serializeVars(vars: MockedVars) {
+    if (vars.sendEmailToRecipeUserId) {
+        vars.sendEmailToRecipeUserId = SuperTokens.convertToRecipeUserId(vars.sendEmailToRecipeUserId.recipeUserId);
+    }
+    if (vars.userPostPasswordReset?.loginMethods.length) {
+        vars.userPostPasswordReset = {
```

Comment:
Isn't this expected to be a proper user object? We are deserialising differently here than elsewhere I think

Reply:
done